test/extended/router/gatewayapicontroller.go

```diff
@@ -370,6 +360,62 @@ func buildGateway(name, namespace, gcname, fromNs, domain string) *gatewayapiv1.
 	}
 }
 
+func waitForLoadbalancerReady(oc *exutil.CLI, gw, gwServiceName string) {
+	// check gateway LB service, note that External-IP might be hostname (AWS) or IP (Azure/GCP)
+	var lbAddress string
+	gwapiClient := gatewayapiclientset.NewForConfigOrDie(oc.AdminConfig())
+	err := wait.PollUntilContextTimeout(context.Background(), 1*time.Second, 10*time.Minute, false, func(context context.Context) (bool, error) {
+		lbService, err := oc.AdminKubeClient().CoreV1().Services("openshift-ingress").Get(context, gwServiceName, metav1.GetOptions{})
+		o.Expect(err).NotTo(o.HaveOccurred())
+		if lbService.Status.LoadBalancer.Ingress[0].Hostname != "" {
+			lbAddress = lbService.Status.LoadBalancer.Ingress[0].Hostname
+		} else {
+			lbAddress = lbService.Status.LoadBalancer.Ingress[0].IP
+		}
+		e2e.Logf("The load balancer External-IP is: %v", lbAddress)
+
+		gwlist, haerr := gwapiClient.GatewayV1().Gateways("openshift-ingress").Get(context, gw, metav1.GetOptions{})
+		e2e.Logf("The gateway hostname address is %v ", gwlist.Status.Addresses[0].Value)
+		o.Expect(haerr).NotTo(o.HaveOccurred())
+		o.Expect(lbAddress).To(o.Equal(gwlist.Status.Addresses[0].Value))
+		return true, nil
+	})
+	o.Expect(err).NotTo(o.HaveOccurred(), "The LB service is not either up nor same with the gateway hostname address")
+}
+
+// assertDNSRecordStatus polls until the DNSRecord's status in the default operand namespace is True.
+func assertDNSRecordStatus(oc *exutil.CLI, gatewayName string) {
+	// find the DNS Record and confirm its zone status is True
+	gatewayDNSRecord := &operatoringressv1.DNSRecord{}
+	err := wait.PollUntilContextTimeout(context.Background(), 2*time.Second, 10*time.Minute, false, func(context context.Context) (bool, error) {
+		gatewayDNSRecords, err := oc.AdminIngressClient().IngressV1().DNSRecords("openshift-ingress").List(context, metav1.ListOptions{})
```

Comment:
Just in case, to prevent reading from the old DNSRecord if something goes side ways (listing or filtering):
```suggestion
	err := wait.PollUntilContextTimeout(context.Background(), 2*time.Second, 10*time.Minute, false, func(context context.Context) (bool, error) {
	        gatewayDNSRecord := &operatoringressv1.DNSRecord{}
		gatewayDNSRecords, err := oc.AdminIngressClient().IngressV1().DNSRecords("openshift-ingress").List(context, metav1.ListOptions{})
```

Reply:
done